cabal.project

```diff
@@ -1,3 +1,5 @@
+index-state: 2019-11-06T00:00:00Z
```

Comment:
I have no problem with something like this `index-state` as an build output but am much less keen on it as an input.

Either that or have a `cabal` command to update that date to the date of the last Hackage index the user has pulled down if the local one is newer.

Reply:
This is going to be a bit tricky. For reproducability, we need to ensure the solver will always produce the same solution. We can do this only with an index-state. haskell.nix updates the index-state once per day. So ideally you don't want to update this outside of the YYYY-MM-DD part.

Having cabal update this automatically give me ulcurs.